bool detected
float64 x
float64 y
float64 d
